src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -179,10 +185,22 @@ IOptions<FeatureConfiguration> featureConfiguration
 
             if (_enableDropInvalidDicomJsonMetadata)
             {
-                // drop invalid metadata
+
                 foreach (DicomTag tag in storeValidatorResult.InvalidTagErrors.Keys)
                 {
+                    // drop invalid metadata
                     dicomDataset.Remove(tag);
+
+                    InstanceIdentifier instanceIdentifier = new InstanceIdentifier(
+                        dicomDataset.GetString(DicomTag.StudyInstanceUID),
+                        dicomDataset.GetString(DicomTag.SeriesInstanceUID),
+                        dicomDataset.GetString(DicomTag.SOPInstanceUID));
```

Comment:
If you have the dataset, you can do 
```suggestion
                    InstanceIdentifier instanceIdentifier = dicomDataset.ToInstanceIdentifier();
```